packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -95,6 +90,9 @@ function classOrObjectMethod(path: NodePath, callId: Object) {
   // Regardless of whether or not the wrapped function is a an async method
   // or generator the outer function should not be
   node.generator = false;
+
+  // Unwrap the wrapper IIFE's environment so super and this and such still work.
+  path.get("body.body.0.argument.callee.arguments.0").unwrapFunctionEnvironment();
```

Comment:
It's gross garbage :P It's reaching in to the path where we put the IIFE, since we only get access to the path _after_ it is inserted, so we have to insert it then do this gross stuff to get the reference.